sql/2020/22_HTTP_2/number_of_https_sites_not_using_h2_returning_upgrade_http_upgrade_header_containing_h2.sql

```diff
@@ -0,0 +1,31 @@
+# standardSQL
+# Number of HTTPS sites not using HTTP/2 which return upgrade HTTP header containing h2
+CREATE TEMPORARY FUNCTION getUpgradeHeader(payload STRING)
+RETURNS STRING
+LANGUAGE js AS """
+  try {
+    var $ = JSON.parse(payload);
+    var headers = $.response.headers;
+    var st = headers.find(function(e) { 
+      return e['name'].toLowerCase() === 'upgrade'
+    });
+    return st['value'];
+  } catch (e) {
+    return '';
+  }
+""";
+
+SELECT 
+  client,
+  firstHtml,  
+  COUNT(0) AS num_requests
+FROM 
+  `httparchive.almanac.requests`
+WHERE
+  date='2020-08-01' AND 
+  url LIKE "https://%" AND
+  JSON_EXTRACT_SCALAR(payload, "$._protocol") != "HTTP/2" AND
```

Comment:
Ditto from comments on previous query. Filter only for `firstHtml` requests and `COUNTIF`.

Reply:
Ditto for including (and differentiating HTTP/3)

Reply:
@rviscomi @bazzadp I made the following changes:

Added the `COUNTIF(getUpgradeHeader(payload) LIKE "%h2%") AS num_requests` and `Count(0) as total` to the `SELECT`

Added `JSON_EXTRACT_SCALAR(payload, '$._protocol') as http_version` to `SELECT` and `GROUP BY`

Added `JSON_EXTRACT_SCALAR(payload, '$._protocol') != ''` to `WHERE` statement to filter out any null values for `http_version` I think it makes sense to filter out nulls....let me know if you disagree